blockchain/Validation.go

```diff
@@ -2805,7 +2827,50 @@ func IsCurrentForFeeEstimation() bool {
 func CheckInputsFromMempoolAndCache(tx *model.Tx, state *model.ValidationState, cache *utxo.CoinsViewCache,
 	mpool *mempool.Mempool, flags uint32, cacheSigStore bool, txData *model.PrecomputedTransactionData) bool {
 
-	return true
+	// todo AssertLockHeld(cs_main)
+	// pool.cs should be locked already, but go ahead and re-take the lock here
+	// to enforce that mempool doesn't change between when we check the view and
+	// when we actually call through to CheckInputs
+	mpool.Mtx.Lock()
+	defer mpool.Mtx.Unlock()
+
+	if tx.IsCoinBase() {
+		panic("critical error")
+	}
+	for _, txin := range tx.Ins {
+		coin := cache.AccessCoin(txin.PreviousOutPoint)
```

Comment:
the variable name, I think use view is more corret.

Reply:
ok